ide/extexecution.base/src/org/netbeans/api/extexecution/base/ExplicitProcessParameters.java

```diff
@@ -0,0 +1,444 @@
+        if (priorityArguments != null) {
+            a.addAll(priorityArguments);
+        }
+        if (middle != null && !middle.isEmpty()) {
+            a.addAll(middle);
+        }
+        if (arguments != null) {
+            a.addAll(arguments);
+        }
+        return a;
+    }
+    
+    /**
+     * Returns the argument lists merged. Priority arguments (if any) are passed first, followed
+     * by {@code middle} (if any), then (normal) arguments. The method is a convenience to build
+     * a complete command line for the launcher + command + command arguments.
+     * @return combined arguments.
+     */
+    public @NonNull List<String> getAllArguments(@NullAllowed String... middle) {
+        return getAllArguments(middle == null ? Collections.emptyList() : Arrays.asList(middle));
+    }
+
+    /**
+     * Merges ExplicitProcessParameters instructions found in the Lookup. See {@link #buildExplicitParameters(java.util.Collection)}
+     * for more details.
+     * @param context context for the execution
+     * @return merged instructions
+     */
+    @NonNull
+    public static ExplicitProcessParameters buildExplicitParameters(Lookup context) {
+        return buildExplicitParameters(context.lookupAll(ExplicitProcessParameters.class));
+    }
+    
+    /**
+     * Merges individual instruction. 
+     * This method serves as a convenience and uniform ("standard") methods to merge argument lists for process execution. Should be used
+     * whenever a process (build, run, tool, ...) is executed. If the feature diverges, it should document how it processes the
+     * {@link ExplicitProcessParamters}. It is <b>strongly recommended</b> to support explicit parameters in order to allow for 
+     * customizations and automation.
+     * <p>
+     * Processes instructions in the order of {@link Builder#withRank(int)} and appearance. Whenever an item is flagged as
+     * a replacement, all arguments (priority arguments) collected to that point are discarded. Item's arguments (priority arguments)
+     * will become the only ones listed.
+     * <p>
+     * <i>Note:</i> if a replacement instruction and all the following (if any) have {@link #getArguments()} {@code null} (= no change), 
+     * the result will report <b>no change</b>. It is therefore possible to <b>discard all contributions</b> by appending a no-change replacement 
+     * last.
+     * 
+     * @param items individual instructions.
+     * @return combined instructions.
+     */
+    public static ExplicitProcessParameters buildExplicitParameters(Collection<? extends ExplicitProcessParameters> items) {
+        List<? extends ExplicitProcessParameters> all = new ArrayList<>(items);
+        Collections.sort(all, (a, b) -> {
+            ExplicitProcessParameters x;
+            int d = a.rank - b.rank;
+            if (d != 0) {
+                return d;
+            }
+            return all.indexOf(a) - all.indexOf(b);
```

Comment:
Querying list that is just being re-ordered for actual position of elements in middle of re-ordering!? Adventurous!

I guess you can return `0` - probably (I believe) the `sort` algorithm guarantees stability for equal elements.

Reply:
Uh-oh !
Fixed in 247e807afc, relevant test added.